IBAnimatable/SystemMoveInAnimator.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  Created by Tom Baranes on 02/04/16.
+//  Copyright © 2016 Jake Lin. All rights reserved.
+//
+
+import UIKit
+
+public class SystemMoveInAnimator: NSObject, AnimatedTransitioning {
+  // MARK: - AnimatorProtocol
+  public var transitionAnimationType: TransitionAnimationType
+  public var transitionDuration: Duration = defaultTransitionDuration
+  public var reverseAnimationType: TransitionAnimationType?
+  public var interactiveGestureType: InteractiveGestureType?
+  
+  // MARK: - private
+  private var fromDirection: TransitionFromDirection
+  
+  init(fromDirection: TransitionFromDirection, transitionDuration: Duration) {
+    self.fromDirection = fromDirection
+    self.transitionDuration = transitionDuration
+    
+    switch fromDirection {
+    case .Left:
+      self.transitionAnimationType = .SystemMoveIn(direction: .Left)
+      self.reverseAnimationType = .SystemMoveIn(direction: .Right)
+      self.interactiveGestureType = .PanFromRight
+    case .Right:
+      self.transitionAnimationType = .SystemMoveIn(direction: .Right)
+      self.reverseAnimationType = .SystemMoveIn(direction: .Left)
+      self.interactiveGestureType = .PanFromLeft
+    case .Top:
+      self.transitionAnimationType = .SystemMoveIn(direction: .Top)
+      self.reverseAnimationType = .SystemMoveIn(direction: .Bottom)
+      self.interactiveGestureType = .PanFromTop
```

Comment:
The pop gesture should be reverse as push animation. You are right, I implemented `handleGesture` in `PanInteractiveAnimator` wrongly, please merge this PR when you free ready, I fix them all in next PR. Thanks.